### PR TITLE
codestyle: Add type annotations to crypto.py and address issues

### PR DESCRIPTION
**Pull request by stefanberger**
Add type annotations to crypto.py and address issues detected by pyright.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>

### CODE REVIEW

File: keylime/tenant.py [hunk 1]
```diff
@@ -1023,7 +1024,10 @@ def do_quote(self):
         logger.info("Quote from %s validated", self.agent_ip)
 
         # encrypt U with the public key
-        encrypted_U = crypto.rsa_encrypt(crypto.rsa_import_pubkey(public_key), self.U)
+        rsa_public_key = crypto.rsa_import_pubkey(public_key)
+        if not isinstance(rsa_public_key, RSAPublicKey):
+            raise UserError(f"Public key is not an RSA public key byte an {type(rsa_public_key).__name__}")
```
Comment (stefanberger) [inline keylime/tenant.py:1029]: typo here!

File: keylime/revocation_notifier.py [hunk 2]
```diff
@@ -182,6 +183,8 @@ def process_revocation(revocation, callback, cert_path):
         logger.warning("Unable to check signature of revocation message: %s not available", cert_path)
     elif "signature" not in revocation or revocation["signature"] == "none":
         logger.warning("No signature on revocation message from server")
+    elif not isinstance(cert_key, RSAPublicKey):
+        logger.warning("Public key from certificate is not an RSA key")
```
Comment (stefanberger) [inline keylime/revocation_notifier.py:187]: Should display type of bad key here as well.

File: keylime/crypto.py [hunk 3]
```diff
@@ -207,7 +207,12 @@ def decrypt(ciphertext, key):
     return decryptor.update(ciphertext) + decryptor.finalize()
 
 
-def generate_selfsigned_cert(name, key, valid_until, other_ips: Optional[List] = None) -> x509.Certificate:
+def generate_selfsigned_cert(
+    name: str,
+    key,  # CERTIFICATE_PRIVATE_KEY_TYPES
```
Comment (stefanberger) [inline keylime/crypto.py:212]: I think this could be an RSAPrivateKey

File: keylime/cloud_verifier_common.py [hunk 4]
```diff
@@ -207,10 +209,13 @@ def prepare_v(agent):
         b64_encrypted_V = agent["b64_encrypted_V"]
         logger.debug("Re-using cached encrypted V")
     else:
+        rsa_public_key = crypto.rsa_import_pubkey(agent["public_key"])
+        if not isinstance(rsa_public_key, RSAPublicKey):
```
Comment (aplanas) [inline keylime/cloud_verifier_common.py:213]: I am not a fan of type hints in Python, but this assert should not be in  [rsa_import_pubkey](https://github.com/keylime/keylime/blob/b6efdf5a1695ae83b1ad271af489ebe1c0e191a2/keylime/crypto.py#L44)?

File: keylime/cloud_verifier_common.py [hunk 5]
```diff
@@ -303,6 +308,8 @@ def prepare_error(agent, msgtype="revocation", event=None):
     # also need to load up private key for signing revocations
     if agent["revocation_key"] != "":
         signing_key = crypto.rsa_import_privkey(agent["revocation_key"])
+        if not isinstance(signing_key, RSAPrivateKey):
```
Comment (aplanas) [inline keylime/cloud_verifier_common.py:311]: Same?

Comment (stefanberger) [reply keylime/cloud_verifier_common.py:311]: Right.

File: keylime/cloud_verifier_common.py (hunk 4 above)
Comment (stefanberger) [reply keylime/cloud_verifier_common.py:213]: You are right. I moved it there.

I think the type hints will help a lot...

**Review by aplanas — APPROVED** [review]

**Review by THS-on — APPROVED** [review]

**Pull request merged**